src/databases/Xdmf/avtXdmfFileFormat.C

```diff
@@ -1615,7 +1660,16 @@ void avtXdmfFileFormat::PopulateDatabaseMetaData(avtDatabaseMetaData *md, int ti
             mt = AVT_RECTILINEAR_MESH;
         }
 
-        AddMeshToMetaData(md, grid->GetName(), mt, extents, nblocks, block_origin, spatial_dimension, topological_dimension);
+        if (mt == AVT_UNKNOWN_MESH) {
+            // This is really a faux mesh object for an associated curve object
+            mt = AVT_RECTILINEAR_MESH;
+            avtMeshMetaData *mmd = new avtMeshMetaData(grid->GetName(), 1, 0, 0, 0, 2, 1, mt);
```

Comment:
What are all these numbers?

Reply:
Huh? :confused: They are some zeros, some ones and a two. There are plenty more examples in various plugins...

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/VASP/avtXDATFileFormat.C#L140-L143

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/ADIOS/avtLAMMPSFileFormat.C#L155-L157

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/NETCDF/avtCCSMReader.C#L277-L279

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/XYZ/avtXYZFileFormat.C#L165-L167

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/Xolotl/avtXolotlFileFormat.C#L871

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/databases/CGNS/avtCGNSFileReader.C#L1169-L1170

In all seriousness, they are the arguments to one of the few ways we support to define metadata for a mesh object in the constructor such as...

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/avt/DBAtts/MetaData/avtMeshMetaData.code#L81-L83

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/avt/DBAtts/MetaData/avtMeshMetaData.code#L187-L188

Reply:
> Sorry. I know what they are being used for. I don't know what they _mean_. 

I get that. You might have to look deeper into the source code to find the answers you need. For example,

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/avt/DBAtts/MetaData/avtMeshMetaData.code#L194-L205

**That aside, other than explaining you don't like *magic*, what are you proposing I do here?**

Before you answer, here's my problem with your problem. This particular case exists in numerous places in the code already. Am I making matters worse adding another case?

Did I wind up having to dive into the source code myself to find the meaning of each of those args? Sure did. I guess I accepted that long ago and haven't looked back. I guess I assume everyone else will have to do the same thing.

Now, if C++ could handle something like...

```
new avtMeshMetaData(name=grid->GetName(), numBlocks=1, blockOrigin=0, cellOrigin=0,
    groupOrigin=0, spatialDimensioin=2, topologicalDimension=1, meshType=mt);
```

I'd be game going forward to do it that way.

Something that comes kinda sorta close to that I guess is this...

```
    {
        std::string const &gridName = grid->Name();
        int const numBlocks = 1, blockOrigin = 0, cellOrigin = 0, groupOrigin = 0;
        int const spatialDimension = 2, topologicalDimension = 1;
        avtMeshType const meshType = mt;
        new avtMeshMetaData(gridName, numBlocks, blockOrigin, cellOrigin,
            groupOrigin, spatialDimension, topologicalDimension, meshType);
    }
```

But I think the enclosing scope is *mandatory* just in case the surrounding code is using the same variable names. But, that could lead to compilers compalining about shadow/aliasing issues too if so and then you have to get more creative with the variable names.

Or, maybe you'd rather I add a new constructor for an `avtMeshMetaData()` object that just sorta does this...

```

// Fasttrack mesh construction cases...
// Here STD means 0 origin, 1BLK means single block
enum avtMeshTypeToConstruct {
    AVT_STD_1BLK_1xN_RECT_MESH_AS_CURVE = 0,
    AVT_STD_1BLK_3D_POINT_MESH = 1, // could be used in LAMMPS and XYZ plugin for example
    .
    .
    .
};

avtMeshMetaData::avtMeshMetaData(std::string meshName, avtMeshTypeToConstruct = mttc)
{
    switch (mttc)
    {
        case AVT_STD_1BLK_1xN_RECT_MESH_AS_CURVE:
        {
            int const numBlocks = 1, blockOrigin = 0, cellOrigin = 0, groupOrigin = 0;
            int const spatialDimension = 2, topologicalDimension = 1;
            avtMeshType const meshType = AVT_RECTILINEAR_MESH;
            return new avtMeshMetaData(meshName, numBlocks, blockOrigin, cellOrigin,
                groupOrigin, spatialDimension, topologicalDimension, meshType);
        }
        case AVT_STD_1BLK_3D_POIONT_MESH:
        {
            int const numBlocks = 1, blockOrigin = 0, cellOrigin = 0, groupOrigin = 0;
            int const spatialDimension = 3, topologicalDimension = 0;
            avtMeshType const meshType = AVT_POINT_MESH;
            return new avtMeshMetaData(meshName, numBlocks, blockOrigin, cellOrigin,
                groupOrigin, spatialDimension, topologicalDimension, meshType);
        }
        .
        .
        .
    }
}
```

and then in the plugin code, we'd have

```
    new avtMeshMetaData(grid->Name(), AVT_STD_1BLK_1xN_RECT_MESH_AS_CURVE);
```

Reply:
> I get that. You might have to look deeper into the source code to find the answers you need. For example,

https://github.com/visit-dav/visit/blob/0f8a9656623e6dae5cabda060e528a8bec76f859/src/avt/DBAtts/MetaData/avtMeshMetaData.code#L194-L205

I understand. I don't think we need to go back and change everything in VisIt. But if we add any new object, I don't see any reason to have names like `nb`, `bo`, `co`, etc. We should never have names like that anywhere in VisIt or in any other project, IMO. VisIt has lived forever so I don't advocate going back and changing names for the countless objects we have, but I hate seeing names like that and going down the rabbit hole to find out what they could mean. I think it should be apparent from reading code what it means.

Look at these horrible names:

https://github.com/visit-dav/visit/blob/2b36a3ee54d21df9f9166207c33e653f44f3bda5/src/avt/Pipeline/Data/avtSpecies.C#L18-L30

Any code that uses these is not readable.

So when I create an `avtSpecies`, I do it like this:

https://github.com/visit-dav/visit/blob/2b36a3ee54d21df9f9166207c33e653f44f3bda5/src/databases/Blueprint/avtBlueprintFileFormat.C#L3237-L3254

> Before you answer, here's my problem with your problem. This particular case exists in numerous places in the code already. Am I making matters worse adding another case?

I would argue yes. Just because there are many parts of VisIt that have code that is hard to read does not mean we should add more.

> But I think the enclosing scope is mandatory just in case the surrounding code is using the same variable names. But, that could lead to compilers compalining about shadow/aliasing issues too if so and then you have to get more creative with the variable names.

If you're using constants, the enclosing scope is unlikely to be necessary, since next time you try to assign a value to one of those names, you will get a compiler error. Adding scope doesn't hurt all that much IMO.

> **That aside, other than explaining you don't like *magic*, what are you proposing I do here?**

I would suggest you use the pattern I linked above where I created an `avtSpecies` object. I create new constants where it makes sense and then I add comments to each thing being passed to the constructor that make it clear what each argument means. Comments can only make things better.